mono/mini/interp/interp.c

```diff
@@ -153,6 +162,11 @@ frame_stack_alloc_ovf (FrameStack *stack, int size, StackFragment **out_frag)
 		current = stack->current = current->next;
 		current->pos = (guint8*)&current->data;
 	} else {
+		StackFragment *tmp = current->next;
+		/* avoid linking to be freed fragments, so the GC can't trip over it */
+		current->next = NULL;
```

Comment:
Volatile? I guess not.

Reply:
Hmm, I guess we should do a `mono_compiler_barrier ();` after setting it to `NULL`? I'm never sure about this stuff.

Reply:
Yes. We should have a compiler barrier here. That should do the trick

Reply:
Currently, the interpreter code doesn't contain barriers, because its too expensive. Instead, the GC does a process wide memory barrier.

Reply:
@vargaz are you suggesting we should remove all the barriers around that stack fragmentation code?

Reply:
But, compiler barrier?

Reply:
ah, k.